packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -1003,6 +1017,14 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
           this._clearPendingInfo();
           return;
         }
+
+        // else if the pending option contains the pending value and we have auto complete and auto full string
+        // update the state
```

Comment:
Some comment about adding functionality for when allowFreeform is false.

Also, I don't think you would need this if the value in the input was updated to include the pending value (it should be handled by the above if check).

Reply:
Are you saying you want this functionality when allowFreeForm is false? Currently there are no changes.